From a751668981a1930523885ee6a8d7905e802baf1d Mon Sep 17 00:00:00 2001
From: mtk20656 <chank.chen@mediatek.com>
Date: Mon, 24 Jul 2023 11:30:27 +0800
Subject: [PATCH 037/149] mtk: hostapd: add extension IE list for non-inherit
 IE in mbssid

Certain clients do not scan all non tx profiles due to absence of
element ID extension list which is mandatory field in non inheritance
IE. Non inheritance Element ID is followed by extension element ID.
Length is expected to be mentioned. Currently we do not support any
extension element and hence filling length as 0.

Signed-off-by: mtk20656 <chank.chen@mediatek.com>
---
 src/ap/ieee802_11.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
 mode change 100644 => 100755 src/ap/ieee802_11.c

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
old mode 100644
new mode 100755
index ac1f940a5..c9f45709d
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -8207,7 +8207,7 @@ static size_t hostapd_eid_mbssid_elem_len(struct hostapd_data *hapd,
 		else if (hapd->conf->xrates_supported)
 			ie_count++;
 		if (ie_count)
-			nontx_profile_len += 4 + ie_count;
+			nontx_profile_len += 5 + ie_count;
 
 		if (len + nontx_profile_len > 255)
 			break;
-- 
2.45.2

